frontend/src/App.js

```diff
@@ -10,11 +11,32 @@ import { V6Chart } from "./components/charts/V6";
 import { V7Chart } from "./components/charts/V7";
 import { V8Chart } from "./components/charts/V8";
 import N1 from "./components/N1";
+import Auth from "./components/Auth";
 import Delete from "./components/Delete";
+import React, { useEffect } from "react";
 
 function App() {
+  useEffect(() => {
+    //timer to logout user after 10 seconds(for testing) of inactivity
+    let timer;
+    const reset = () => {
+      if (localStorage.getItem("token")) {
+        clearInterval(timer);
+        timer = setInterval(() => {
+          localStorage.clear();
+          window.location.href = "/login";
+        }, 10000);
+      }
+    };
+    document.onkeydown = reset;
+    document.onmousemove = reset;
+  }, []);
```

Comment:
Maybe remove the inactivity timer as that does not invalidate the JWT. The JWT itself is valid for 1 hour so we could just check for that and logout if it has expired.